chain/bitcoind.go

```diff
@@ -867,7 +910,16 @@ func (c *BitcoindClient) rescan(hash *chainhash.Hash) error {
 // any matching transactions, sending notifications along the way.
 func (c *BitcoindClient) filterBlock(block *wire.MsgBlock, height int32,
 	notify bool) ([]*wtxmgr.TxRecord, error) {
-	log.Debugf("Filtering block %d (%s)\n", height, block.BlockHash())
+	// If we're earlier than wallet birthday, don't do any notifications.
+	c.clientMtx.RLock()
+	startTime := c.startTime
+	c.clientMtx.RUnlock()
+	if block.Header.Timestamp.Before(startTime) {
```

Comment:
Doesn't this need to check the block for nil-ness? As in the logic above, if we're _before_ the wallet's birthday, then the block above will be nil.

Reply:
Ah nvm, I see it fetches a dummy block.